### PR TITLE
nfs: move repository configuration in ceph-nfs role

### PR DESCRIPTION
**Pull request by guits**
This is something that has nothing to do in `ceph-common`, this
is too specific to `ceph-nfs` role.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

File: roles/ceph-iscsi-gw/tasks/prerequisites.yml [hunk 1]
```diff
@@ -1,4 +1,40 @@
 ---
+- name: set_fact ceph_iscsi_repos
+  set_fact:
+    ceph_iscsi_repos:
+      - ceph-iscsi-config
+      - tcmu-runner
+  when:
+    - ceph_origin == 'repository'
```
Comment (leseb) [inline roles/ceph-iscsi-gw/tasks/prerequisites.yml]: `ansible_os_family == 'RedHat'`?

File: roles/ceph-iscsi-gw/tasks/prerequisites.yml [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+    - ceph_iscsi_config_dev
+
+- name: fetch ceph-iscsi-config red hat development repository
+  uri:
+    url: https://shaman.ceph.com/api/repos/{{ item }}/{{ ceph_dev_branch }}/{{ ceph_dev_sha1 }}/{{ ansible_distribution | lower }}/{{ ansible_distribution_major_version }}/repo
+    return_content: yes
+  register: ceph_iscsi_config_dev_yum_repo
+  with_items: "{{ ceph_iscsi_repos }}"
+  when:
+    - ceph_origin == 'repository'
```
Comment (leseb) [inline roles/ceph-iscsi-gw/tasks/prerequisites.yml]: ditto

File: roles/ceph-iscsi-gw/tasks/prerequisites.yml [hunk 3]
```diff
@@ -1,4 +1,40 @@
...
+    content: "{{ item.0.content }}"
+    dest: "/etc/yum.repos.d/{{ item.1 }}-dev.repo"
+    owner: root
+    group: root
+    backup: yes
+  with_together:
+    - "{{ ceph_iscsi_config_dev_yum_repo.results }}"
+    - "{{ ceph_iscsi_repos }}"
+  when:
+    - ceph_origin == 'repository'
```
Comment (leseb) [inline roles/ceph-iscsi-gw/tasks/prerequisites.yml]: ditto

File: roles/ceph-nfs/tasks/pre_requisite_non_container.yml [hunk 4]
```diff
@@ -1,4 +1,79 @@
...
+    gpgcheck: yes
+    state: present
+    gpgkey: "{{ ceph_stable_key }}"
+    baseurl: "{{ ceph_mirror }}/nfs-ganesha/rpm-{{ nfs_ganesha_stable_branch }}/{{ ceph_release }}/$basearch"
+  when:
+    - ansible_os_family == 'RedHat'
+    - nfs_ganesha_stable
+    - ceph_origin == 'repository'
+    - ceph_repository == 'custom'
+
```
Comment (leseb) [inline roles/ceph-nfs/tasks/pre_requisite_non_container.yml]: remove extra line

Comment (guits) [reply roles/ceph-nfs/tasks/pre_requisite_non_container.yml]: done

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Pull request merged**